gnosis_scan_api.py

```diff
@@ -0,0 +1,84 @@
+import requests
+import json
+import datetime
+import calendar
+from constants import REQUEST_TIMEOUT, SUCCESS_CODE
+
+
+def get_block_range(year, month, day):
+    date_time = datetime.datetime(year, month, day)
+    start_timestamp = int(calendar.timegm(date_time.timetuple()))
+    url = (
+        "https://api.gnosisscan.io/api?module=block&action=getblocknobytime"
+        + "&timestamp="
+        + str(start_timestamp)
+        + "&closest=after&apikey=YourApiKeyToken"
+    )
+    try:
+        response = requests.get(
+            url,
+            timeout=REQUEST_TIMEOUT,
+        )
+        if response.status_code == SUCCESS_CODE:
+            data = json.loads(response.text)
+            if data["status"] == "1":
+                start_block = int(data["result"])
+            else:
+                print("Start timestamp. " + data["result"])
+                exit()
+    except Exception as e:
+        print(f"Fetching block range failed with error: {e}")
+        raise e
+
+    end_timestamp = start_timestamp + 7 * 24 * 60 * 60 - 1
+    url = (
+        "https://api.gnosisscan.io/api?module=block&action=getblocknobytime"
+        + "&timestamp="
+        + str(end_timestamp)
+        + "&closest=before&apikey=YourApiKeyToken"
+    )
+    try:
+        response = requests.get(
+            url,
+            timeout=REQUEST_TIMEOUT,
+        )
+        if response.status_code == SUCCESS_CODE:
+            data = json.loads(response.text)
+            if data["status"] == "1":
+                end_block = int(data["result"])
+            else:
+                print("End timestamp. " + data["result"])
+                exit()
+    except Exception as e:
```

Comment:
Combining the comments above, this could be written as (up to formatting)
```suggestion
        response.raise_for_status()
        data = response.json()
        if data["status"] != "1":
            raise requests.exceptions.RequestException(f"Request failed with status {data['status']}, end timestamp {data['result']}")
        end_block = int(data["result"])
    except requests.exceptions.RequestException as e:
```